pkg/querier/queryrange/codec.go

```diff
@@ -585,7 +609,55 @@ func (Codec) DecodeResponse(_ context.Context, r *http.Response, req queryrangeb
 	}
 }
 
-func (Codec) EncodeResponse(ctx context.Context, res queryrangebase.Response) (*http.Response, error) {
+func decodeResponseProtobuf(r *http.Response, req queryrangebase.Request) (queryrangebase.Response, error) {
+	var buf []byte
```

Comment:
Ok, I'll preallocate the body length for starters. The original code has the same issue.

Reply:
Right, but we're allocating memory for every `io.ReadAll` which is going to be discarded after each request. I'm not sure if this will escape to the heap, but it could because the size of the body is not known apriori i think?

Can you add a benchmark for this?